packages/rag/src/tools/vector-query.ts

```diff
@@ -25,13 +25,27 @@ export const createVectorQueryTool = ({
 }): ReturnType<typeof createTool> => {
   const toolId = id || `VectorQuery ${vectorStoreName} ${indexName} Tool`;
   const toolDescription = description || defaultVectorQueryDescription();
+  
+  // Create base schema with required fields
+  const baseSchema = {
+    queryText: z.string().describe('The text query to search for in the vector database'),
+    topK: z.number().describe(topKDescription),
+  };
+  
+  // Add filter field based on enableFilter setting
+  const inputSchema = enableFilter
+    ? z.object({
+        ...baseSchema,
+        filter: z.string().describe(filterDescription),
+      })
+    : z.object({
+        ...baseSchema,
+        filter: z.string().optional().describe(filterDescription),
+      });
```

Comment:
After playing around with different models, I think we want to do something like this:

```suggestion
const baseSchema = {
    queryText: z.string().describe('The text query to search for in the vector database'),
    topK: z.coerce.number().describe(topKDescription),
  };
  const inputSchema = enableFilter
    ? z.object({
        ...baseSchema,
        filter: z.string().describe(filterDescription),
      })
    : z.object({
        ...baseSchema,
      });
```